### DISCUSSION TITLE
TeslaMate stopped collecting data after power outage at home

### DISCUSSION THREAD

**Discussion by xeoMe015** [teslamate-org/teslamate#3845] [Q&A] [answered]
Hi All,

I've been running Teslamate and AdGuard on my raspberry pi for months without issues using docker.
on 8th of April i had a power outage at home and now my TeslaMate is not collecting data anymore.

I logged out and try logging in again but it keeps on failing.

Teslamate logs:

POST https://auth.tesla.com/oauth2/v3/token -> error: "non-existing domain" (8135.906 ms)

Token refresh failed: %TeslaApi.Error{
pi-teslamate-1 |   reason: :token_refresh,
pi-teslamate-1 |   message: "non-existing domain",
pi-teslamate-1 |   env: nil

I think i have adjusted to DNS servers to localhost and 8.8.8.8 and restarted everyting but still no succes.
I'm not guru in raspberry and docker so any advice is much appreciated.

**Comment by cwanja** [comment]
Without knowing how your local network is setup, localhost could still go through AdGuard. 

Shut down AdGuard, shut down TeslaMate, adjust your DNS to 1.1.1.1 and restart TeslaMate. See if that works. 

If that does work, start up AdGuard. If TeslaMate stops working, AdGuard is your issue. You could probably whitelist all Tesla domains via a wildcard as well.

**Comment by xeoMe015** [comment, marked as answer]
Steps i have followed.

sudo nano /etc/dhcpcd.conf
have added "static domain_name_server=8.8.8.8

Restarted the DNS services.
sudo service dhcpcd restart

Removed the containers for teslamate
docker ps
docker stop "container names"

recreated the containers with:
docker compose up -d

Everything seems to be working now i think. (Not sure if my dns entries are cached but i believe so)

**Reply by xeoMe015** [reply]
Seems like it is working for a short period of time and then stops again.
Very anoying because i love to read out data from my Tesla but Adguard has even more priority.

**Reply by cwanja** [reply]
If you shut down AdGuard, does TeslaMate work? Review the logs in your AdGuard and see if Tesla is being blocked. If so, whitelist Tesla domains.

**Reply by xeoMe015** [reply]
@cwanja Previously it has always worked but since the power outage it is a bit weird.
Currently it seems to be working but i've noticed a few things from the overview.

Offline = sleeping (it seems) 

My AdGuard seems to be working and TeslaMate also seems to be working but i'm not sure.
I will do a test drive tonight and see if the trip is being stored in TeslaMate.
Still gives a health check error message in the teslamate configuration though.

**Reply by cwanja** [reply]
> Offline = sleeping (it seems)

See #3753.

**Reply by xeoMe015** [reply]
Everything is working as expected again.
